### ISSUE TITLE
OAR::IO::get_gantt_jobs_to_launch question

### ISSUE THREAD

**Issue by npf** [oar-team/oar#47]
mysql query uses m.moldable_index = \'CURRENT\', while pgsql doesn't. Looking at OAR::IO::get_gantt_hostname_to_wake_up, there is no such difference.

Bug?

**Comment by npf** [comment]
Also, I don't understant while the function needs to be 2 folds, with 2 requests.

Could we do:
```
SELECT DISTINCT(j.job_id)
                FROM gantt_jobs_resources g1, gantt_jobs_predictions g2, jobs j, moldable_job_descriptions m, resources
                WHERE
                   g1.moldable_job_id = g2.moldable_job_id
                   AND m.moldable_id = g1.moldable_job_id
                   AND j.job_id = m.moldable_job_id
                   AND g2.start_time <= $date
                   AND j.state = \'Waiting\'
                   AND resources.resource_id = g1.resource_id
                   AND resources.state = \'Alive\'
                   AND resources.next_state NOT IN (\'Dead\',\'Suspected\',\Absent\')

**Comment by npf** [comment]
[removed previous comment]

In one request, I think we could do:
```
SELECT g2.moldable_job_id, g1.resource_id, j.job_id
                FROM gantt_jobs_resources g1, gantt_jobs_predictions g2, jobs j, moldable_job_descriptions m, resources r
                WHERE
                    m.moldable_index = \'CURRENT\'
                    AND g1.moldable_job_id = g2.moldable_job_id
                    AND m.moldable_id = g1.moldable_job_id
                    AND j.job_id = m.moldable_job_id
                    AND g2.start_time <= $date
                    AND j.state = \'Waiting\'
                    AND r.resource_id = g1.resource_id
                    AND r.state = \'Alive\'
                    AND NOT EXISTS ( 
                        SELECT 1 
                        FROM resources rr 
                        WHERE rr.resource_id = g1.resource_id 
                            AND ( rr.state IN (\'Dead\',\'Suspected\',\Absent\')
                            OR rr.next_state IN (\'Dead\',\'Suspected\',\Absent\') )
                    )
```
To test.

**Comment by npf** [comment]
[updated the request above]

**Comment by npf** [comment]
As reported by @capitn, having "m.moldable_index = \'CURRENT\'" in the where clause actually slows down pgsql.
Fixed in 3e60532cc6109cc7705def6f0b4836013ed8bea8, but use the new code with the single query (with "exists in").
